### PR TITLE
Please help - it's been a long day

### PR DESCRIPTION
**Pull request by marchampson**
Here's the output of http://news.local:

Notice: A session had already been started - ignoring session_start() in /Users/marchampson/Sites/masterclass-repo/public/Index.php on line 3

Warning: Invalid argument supplied for foreach() in /Users/marchampson/Sites/masterclass-repo/Upvote/Library/Front/Controller.php on line 50

Notice: Undefined index: call in /Users/marchampson/Sites/masterclass-repo/Upvote/Library/Front/Controller.php on line 31

Warning: array_shift() expects parameter 1 to be array, null given in /Users/marchampson/Sites/masterclass-repo/Upvote/Library/Front/Controller.php on line 32

Warning: array_shift() expects parameter 1 to be array, null given in /Users/marchampson/Sites/masterclass-repo/Upvote/Library/Front/Controller.php on line 33

Fatal error: Class '' not found in /Users/marchampson/Sites/masterclass-repo/Upvote/Library/Front/Controller.php on line 34

### CODE REVIEW

File: Upvote/Application/Controller/Comment.php [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?php
+namespace Application\Controller;
```
Comment (brandonsavage) [inline Upvote/Application/Controller/Comment.php:2]: Your namespaces are technically wrong, but I'm guessing you set the include path to start from Upvote.

File: Upvote/Library/Front/Controller.php [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
+namespace Library\Front;
 
-class MasterController {
+class Controller {
     
     private $config;
     
     public function __construct($config) {
         $this->_setupConfig($config);
+		set_include_path('/Users/marchampson/Sites/masterclass-repo/Upvote');
```
Comment (brandonsavage) [inline Upvote/Library/Front/Controller.php:10]: Yep, that's what you did.

Comment (brandonsavage) [inline Upvote/Library/Front/Controller.php:10]: For abstraction's sake I'd throw this in index.php.

File: Upvote/Library/Front/Controller.php [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
+		spl_autoload_register(array($this, 'autoload'));
     }
     
+	function autoload($className)
```
Comment (brandonsavage) [inline Upvote/Library/Front/Controller.php:14]: Best practice is to always have a visibility keyword (even when it's public).

File: config.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (brandonsavage) [inline config.php:1]: You'll want to add the config.php file to your .gitignore and commit the config.php-init instead.

File: public/.htacces [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<IfModule mod_rewrite.c>
```
Comment (brandonsavage) [inline public/.htacces:1]: Same thing as the config.php.

File: Upvote/Application/Controller/Comment.php (hunk 1 above)
Comment (marchampson) [inline Upvote/Application/Controller/Comment.php:2]: I did, but have changed it to match yours now.

File: Upvote/Library/Front/Controller.php (hunk 2 above)
Comment (marchampson) [inline Upvote/Library/Front/Controller.php:10]: Noted and done. Thanks.

File: config.php (hunk 4 above)
Comment (marchampson) [inline config.php:1]: Good point, the mysql passwords are unsafe as it is :)

**Pull request merged**